### DISCUSSION TITLE
Ambient pressure sensor

### DISCUSSION THREAD

**Discussion by Hein-NonesensE** [joeycastillo/Sensor-Watch#82] [Q&A]
Have you thought about the integration of a pressure sensor? Does anyone know one that might fit? It would be useful for several faces from diving to hiking. I know it might not be feasible because of the watertight enclosure, but then air isn't water. If it doesn't work, there's of course still the possibility to hack the enclosure.

Faces that came to my mind that could use a pressure sensor:

Sailing:
- Current ambient pressure
- Pressure log (like temperature log)
- Pressure change (1h, 3h, potentially with configurable alarm for fast falling pressure)

Hiking, Mountainbiking:
- Barometric height
- Cumulated traveled height since reset

Diving (not that I'd recommend it):
- Depth
- Depth change 

Unfortunately I'm not a programmer, I just hacked my way through some arduino projects for lab use and stuff like that. So I don't know if I'd be able to help with anything above except soldering in a sensor and testing.
On a side note, I'd also like a regatta face (which of course needs no sensors at all). If no fellow sailor takes this up, I might try how far I come with that in winter as a start.

**Comment by joeycastillo** [comment]
Interestingly the first sensor board I designed for the watch had a pressure sensor, although I never wrote code to read pressure data (I was more interested in temperature). In [the Sensor Boards folder](https://github.com/joeycastillo/Sensor-Watch/tree/main/PCB/Sensor%20Boards), both OSO-MISC-21-001 and OSO-MISC-21-002 incorporate the Bosch BME280, a combined temperature, humidity and barometric pressure sensor. 21-002 is a bit easier to build; 21-001 had more and smaller components, and I never got it working before deciding to use thermistors instead. 

Still, the OSO-MISC-21-002 board with BME280 board absolutely did work; there's [demo code for reading temperature and humidity here](https://github.com/joeycastillo/Sensor-Watch/tree/616587a2030be054277ef74fdbf0be626684db3c/apps/Sensor%20Watch%20BME280%20Project), although nothing was particularly well structured at that point. As I recall this app was reading temperature reliably, and getting raw data for humidity even if my conversion math was a bit glitchy. I never bothered reading pressure data, but there's a definition in the header for the pressure register; it probably wouldn't be too hard to implement. How well you can read pressure from inside the watertight case, I don't know — but it's worth a shot!

**Reply by Hein-NonesensE** [reply]
That's a pleasant surprise! Can't wait to get my hands on a sensor watch and see how it works out. Unfortunately, I'll have to wait for quite a while as I just ordered the day I made this post, but if in the meantime you work on this, I'll be delighted to read about it.

**Reply by casundra** [reply]
I'm interested in playing with a barometric pressure sensor as well - just going off of what I can currently get my hands on, I think the [BMP388 ](https://www.digikey.com/en/products/detail/bosch-sensortec/BMP388/8322638)or the[ TDK ICP-10101](https://www.digikey.com/en/products/detail/tdk-invensense/ICP-10101/9555751) are interesting.  The latter is particularly low power, which I think might be nice for the watch.  @joeycastillo - do I recall you made a dev board for the Sensor Watch that you could use to test out new sensors before making custom boards and putting them in the actual watch?  Adafruit has a dev board for the BMP388 so if I had a way to hook it up, I could start playing with it pretty fast.  (and yes, the best way to procrastinate on one's own backlog of projects is clearly to work on someone else's).  :)

**Reply by joeycastillo** [reply]
> @joeycastillo - do I recall you made a dev board for the Sensor Watch that you could use to test out new sensors

I had a couple of things... one was just a test points board with larger test points than the ones y'all got ([Sensor Watch Testpoints](https://github.com/joeycastillo/Sensor-Watch/blob/main/PCB/Sensor%20Boards/OSO-MISC-21-012%20Sensor%20Watch%20Testpoints%20v2.brd)), another actually predates the watch and is super weird: it breaks the nine pins out into a JST-SH 9-pin connector, which was compatible with some old sensor boards I was working on, and very much doesn't fit into the watch case ([OSO-MISC-20-026](https://github.com/joeycastillo/Sensor-Cap/blob/main/PCB/Misc/OSO-MISC-20-026%20Ring-Cap%20Adapter.brd)). I think it would make a lot of sense to make a little flex adapter that could adapt the flex connector to STEMMA QT, but I haven't designed that yet; could be a good idea for this week though!

> I'm interested in playing with a barometric pressure sensor as well

I know they're expensive now, but one of the oldest sensor boards, [Sensor Watch Environment](https://github.com/joeycastillo/Sensor-Watch/blob/main/PCB/Sensor%20Boards/OSO-MISC-21-002%20Sensor%20Watch%20Environment.brd) uses the BME280, and has the benefit of some driver code already written. I'd have to dig it out of the git history, but it worked for reading temperature, and almost worked for humidity. There's a barometric pressure sensor in there too, I just hadn't yet written the code for reading it.
